2022/Matricardi/README.md

```diff
@@ -0,0 +1,75 @@
+# Autore
+
+Alessio Matricardi (a.matricardi@studenti.unipi.it)
+
+# Obiettivo
+
+L'obiettivo del progetto è l'implementazione di un sistema di individuazione di nervosismo (skewness) all'interno di serie temporali.
+
+Il sistema deve essere calibrato dall'utente con i primi N valori all'interno della serie temporale, ragion per cui è opportuno che la serie presenti, almeno inizialmente, un comportamento che riteniamo standard. Il processo di calibrazione calcola la deviazione standard degli N valori.
+Da questo momento in poi, il programma cerca il nervosismo all'interno della serie procedendo con la tecnica della finestra mobile: il sistema procede calcolando la deviazione standard sui W valori presenti all'interno della finestra mobile.
+Un nervosismo all'interno della finestra viene individuato se la deviazione standard calcolata è 3 volte maggiore di quella di calibrazione.
+La finestra si sposta ad ogni iterazione di 1/10 della sua grandezza (e.g. se W = 20 -> finestra mobile di 2 elementi per iterazione)
+
+![plot](./graph.png)
+
+# Dipendenze
+
+Le dipendenze sono all'interno del file `requirements.txt`.
+
+Per installarle, se non presenti sulla propria macchina, eseguire da linea di comando:
+```
+pip3 install requirements.txt
+```
+Per evitare l'installazione di dipendenze non volute, è possibile installare ed eseguire il programma all'interno di un ambiente virtuale.
+
+# Come eseguire
+
+Parametri obbligatori:
+
+- `-n` numero di elementi con cui calibrare il sistema
+- `-w` numero di elementi presenti nella finestra mobile
+
+- `--rrd <rrd file>` oppure `--json <json file>` per indicare il file con il segnale da analizzare
+
+Parametri opzionali:
+
+- `-v` verbose mode (mostra log di debug come ad esempio quando non viene rilevata un anomalia)
+- `-p` plotta in nero il segnale ed in rosso dove il segnale è stato riconosciuto come nervoso
+
+# Come testare
+
+Un esempio tipico di esecuzione è
+```
+python3 skewness.py -n 50 -w 40 -p --rrd test/signal.rrd
+```
+
+Il programma è stato testato con tutti i file di test forniti.
```

Comment:
Per fare un test piu' esaustivo vorrei che tu ti installassi [ntopng](https://github.com/ntop/ntopng) gli facessi generare degli RRD e modificassi il tuo tool in modo che vada su questi RRD a vedere se ci sono anomalie e le stampasse. Questo mi permetterebbe di poter analizzare il comportamento del tuo tool in un caso reale e valutare gli eventuali falsi positivi/negativi.